jetty-infinispan/infinispan-common/src/main/java/org/eclipse/jetty/session/infinispan/InfinispanSessionDataSerializer.java

```diff
@@ -0,0 +1,69 @@
+package org.eclipse.jetty.session.infinispan;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+
+import org.infinispan.commons.marshall.Externalizer;
+
+public class InfinispanSessionDataSerializer implements Externalizer<InfinispanSessionData>
+{
+
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public void writeObject(ObjectOutput output, InfinispanSessionData object) throws IOException
+    {
+        output.writeUTF(object.getId()); //session id
```

Comment:
Please see the o.j.e.session.infinispan.SessionDataMarshaller regarding the requirements for the first field that is serialized: the version.

Reply:
Rather than copying the serialization, I'd rather that there was a refactoring that made it possible for your new class and the SessionDataMarshaller to use the same code.